src/main/java/Deadline.java

```diff
@@ -0,0 +1,13 @@
+public class Deadline extends Task {
+    protected String dueBy;
+
+    public Deadline(String description, String dueBy) {
+        super(description);
+        this.dueBy = dueBy;
+    }
+
+    @Override
+    public String getTaskDetails() {
+        return "[D]" + super.getTaskDetails() + " (by: " + dueBy + ")";
+    }
```

Comment:
good use of super() - reduces alot of unnecessary duplicate code

Reply:
Thanks Jun Hao!